### ISSUE TITLE
Search of a split word not finding the non split word.

### ISSUE THREAD

**Issue by alciono** [NixOS/nixos-search#183]
For example:

If I search for "open vpn", without quotes, I would expect to see "openvpn" being part of the retrieved list. 

Instead The list is showing 10 items unrelated to "openvpn". 

This type of search used to work with the previous search version while narrowing the list down to the few items directly related to the word in question without showing un unnecessary long list of unrelated items.

Would it be possible to design a menu which would give the possibility to choose between different ways of searching?

Thank you

**Comment by garbas** [comment]
@alciono Thank you for reporting this. I'm trying to tune the search and it is easy to miss things.

The search should definetly show all 10 openvpn related packages. I think I have an idea why it is not.

**Comment by alciono** [comment]
Thanks a lot garbas for paying attention and taking the time to feed back!
Exiting idea this faceted search. Looking forward to seeing the results.

Please let me know if I could be of any help knowing that I am just starting to scratch the surface of the NIX intricacies.

Cheers

**Comment by garbas** [comment]
@alciono It took a bit longer then expected, but I managed to improve the search (in #197). Would you have time to test it and give feedback?

**Comment by alciono** [comment]
Many thanks garbas. I tried a few similar searches and it is working well now.

I also tried ".httpd." ( without quotes) to isolate from .lighttpd.,  for example, and  get a blank result.
Search doesn't seem to acknowledge punctuation. Is it on purpose?

If this remark is worth considering, would you take it from here or should I open  a new issue? 

Thank you very much for your quick and good work!

**Comment by garbas** [comment]
@alciono if you want to isolate from others options I would use whole prefix `services.httpd.` even `httpd.` works.

In the future this is going to be improved even more with _grouping_ of the options via faceted search. This is actually something I will be focusing next once redesign of nixos.org is done.
